src/@types/polyfill.ts

```diff
@@ -0,0 +1,36 @@
+/*
+Copyright 2020 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+export function polyfillTouchEvent() {
+    // Firefox doesn't have touch events, so create a fake one we can rely on lying about.
+    if (!window.TouchEvent) {
+        // We have no intention of actually using this, so just lie.
```

Comment:
Some comment somewhere should probably mention this is only here to support a specific library... I also worry slightly that this could trigger other unrelated libraries into thinking we suddenly are on a touch device, but since Chrome appears to expose this on non-touch, presumably that is unlikely to happen in practice.

Reply:
There's also `window.supportsTouchDevices` or something which is a more common way to detect touch support. The presence of the event class is mostly for the `instanceof` checks that people run.